### ISSUE TITLE
[BUG] [Regression] [type-safe-api] Build fail if any parameter had '$ref' in Open API model

### ISSUE THREAD

**Issue by v41e** [aws/aws-pdk#590]
### Describe the bug

When using a `$ref` in any parameter in Open API model, build fails.

```
@acme/api-model: running command pnpm exec ts-node parse-openapi-spec.ts --specPath=/Users/johndoe/.../packages/api/model/src/main/openapi/main.yaml --outputPath=/Users/johndoe/.../packages/api/model/.api.json
@acme/api-model: Request parameters must be of type number, integer, boolean, string or arrays of these. Found invalid parameters:
@acme/api-model: get /hello: undefined
@acme/api-model: 👾 Task "build » compile » generate" failed when executing "npx --yes -p @aws/pdk@$AWS_PDK_VERSION type-safe-api.parse-openapi-spec --spec-path src/main/openapi/main.yaml --output-path .api.json" (cwd: /Users/johndoe/.../packages/api/model)
@acme/api-model: Warning: run-commands command "pnpm dlx projen build" exited with non-zero status code
```

It works on a project locked in 0.19.66.

### Expected Behavior

The build doesn't fail, it worked in previous version

### Current Behavior

The build faild

### Reproduction Steps

1. Follow the tutorial in https://aws.github.io/aws-pdk/developer_guides/type-safe-api/index.html to setup the project
2. Instead of this Open API spec file https://aws.github.io/aws-pdk/developer_guides/type-safe-api/using_openapi.html use the following (similar to those in tutorial but with $ref in parameter)

```yaml
openapi: 3.0.3
info:
  version: 1.0.0
  title: Example API
paths:
  /hello:
    get:
      operationId: sayHello
      x-handler:
        language: typescript
      parameters:
        - $ref: '#/components/parameters/HelloId'
      responses:
        '200':
          description: Successful response
          content:
            'application/json':
              schema:
                $ref: '#/components/schemas/HelloResponse'
components:
  parameters:
    HelloId:
      in: query
      name: id
      schema:
        $ref: '#/components/schemas/HelloId'
      required: false
  schemas:
    HelloId:
      type: string
    HelloResponse:
      type: object
      properties:
        id:
          $ref: '#/components/schemas/HelloId'
        message:
          type: string
      required:
        - message

```

### Possible Solution

By checking the source code of the parser, the code have now more logic and more checks

This line tells what are allowed types

https://github.com/aws/aws-pdk/blob/mainline/packages/type-safe-api/scripts/type-safe-api/parser/parse-openapi-spec.ts#L32

There are checks for the type, for arrays, but not for refs so we fall in this error

https://github.com/aws/aws-pdk/blob/mainline/packages/type-safe-api/scripts/type-safe-api/parser/parse-openapi-spec.ts#L118

And then in https://github.com/aws/aws-pdk/blob/mainline/packages/type-safe-api/scripts/type-safe-api/parser/parse-openapi-spec.ts#L128

A check to resolve ref should be add after array check

### PDK version used

0.22.9

### What languages are you seeing this issue on?

Typescript

### Environment details (OS name and version, etc.)

macOS Sonoma 14.0.0 Apple M1 Pro

**Comment by cogwirrel** [comment]
Hey, thanks for reporting this! 😄 

Those `$ref`s should definitely be allowed, and resolved as part of the validation!

I'll send a PR to fix this shortly!

**Comment by cogwirrel** [comment]
The fix has been released in [`0.22.12`](https://github.com/aws/aws-pdk/releases/tag/v0.22.12) now - let me know if you have any further issues!

Thanks,
Jack

**Comment by v41e** [comment]
Hi @cogwirrel,

It now works as expected.

Thanks a lot :)
